aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -888,6 +896,119 @@ module aptos_framework::delegation_pool {
         }
     }
 
+    /// Initialize a delegation pool from an existing owned staking contract.
+    /// The newly created delegation pool will preserve the operator and commission fee. The staker becomes the pool
+    /// owner and has the same delegated voter set.
+    public entry fun initialize_delegation_pool_from_staking_contract(
+        staker: &signer,
+        operator: address,
+    ) acquires DelegationPool, GovernanceRecords, BeneficiaryForOperator, NextCommissionPercentage {
+        assert!(features::delegation_pools_enabled(), error::invalid_state(EDELEGATION_POOLS_DISABLED));
+        assert!(
+            features::staking_contract_to_delegation_pool_conversion_enabled(),
+            error::invalid_state(ESTAKING_CONTRACT_TO_DELEGATION_POOL_CONVERSION_NOT_SUPPORTED)
+        );
+
+        let staker_address = signer::address_of(staker);
+        assert!(!owner_cap_exists(staker_address), error::already_exists(EOWNER_CAP_ALREADY_EXISTS));
+
+        // do not have to withhold the `add_stake` fee from staker if there is no pending_active stake
+        let (_, _, pending_active, _) = stake::get_stake(
+            staking_contract::stake_pool_address(staker_address, operator)
+        );
+        assert!(pending_active == 0, error::invalid_state(EPENDING_ACTIVE_STAKE_ON_STAKING_CONTRACT));
+
+        // destroy staking contract and take ownership of its underlying stake pool
+        let (
+            principal,
+            pool_address,
+            owner_cap,
+            operator_commission_percentage,
+            distribution_pool,
+            stake_pool_signer_cap,
+        ) = staking_contract::destroy_staking_contract(staker, operator);
+
+        // convert existing commission to a 2-decimals precision percentage
+        operator_commission_percentage = operator_commission_percentage * 100;
+
+        let stake_pool_signer = account::create_signer_with_capability(&stake_pool_signer_cap);
+        coin::register<AptosCoin>(&stake_pool_signer);
```

Comment:
Curious - why is this needed?

Reply:
staking contract uses `stake::add_stake_with_cap` so APT is withdrawn from staker signer and deposited directly into the underlying stake pool
delegation pool module uses `stake::add_stake` which withdraws APT from the caller signer, which is the resource account of the delegation pool, so the resource account has to firstly get the APT from delegator signer and thus it requires AptosCoin registration